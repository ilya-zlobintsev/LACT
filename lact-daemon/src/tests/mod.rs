use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
