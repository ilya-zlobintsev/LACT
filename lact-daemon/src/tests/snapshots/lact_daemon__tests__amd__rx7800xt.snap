---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": {
    "max_mclk": 1300,
    "max_sclk": 2660,
    "table": {
      "type": "amd",
      "value": {
        "data": {
          "current_mclk_range": {
            "max": 1300,
            "min": 97
          },
          "current_sclk_range": {
            "max": 2660,
            "min": 500
          },
          "od_range": {
            "curve_sclk_points": [],
            "curve_voltage_points": [],
            "mclk": {
              "max": 1500,
              "min": 97
            },
            "sclk": {
              "max": 5000,
              "min": 500
            },
            "sclk_offset": null,
            "voltage_offset": {
              "max": 0,
              "min": -450
            }
          },
          "sclk_offset": null,
          "vddc_curve": [],
          "voltage_offset": -100
        },
        "kind": "rdna"
      }
    }
  },
  "info": {
    "driver": "amdgpu",
    "link_info": {
      "current_speed": "16.0 GT/s PCIe",
      "current_width": "16"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "Navi 32 [Radeon RX 7700 XT / 7800 XT]",
        "model_id": "747E",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      },
      "subsystem_pci_info": {
        "model": "PowerColor RX 7800 XT Hellhound",
        "model_id": "2427",
        "vendor": "Tul Corporation / PowerColor",
        "vendor_id": "148C"
      }
    },
    "vbios_version": "113-EXT90249-100",
    "vulkan_info": {
      "api_version": "",
      "device_name": "",
      "driver": {
        "version": 0
      },
      "enabled_layers": [],
      "extensions": {},
      "features": {}
    }
  },
  "pci_info": {
    "device_pci_info": {
      "model": "Navi 32 [Radeon RX 7700 XT / 7800 XT]",
      "model_id": "747E",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    },
    "subsystem_pci_info": {
      "model": "PowerColor RX 7800 XT Hellhound",
      "model_id": "2427",
      "vendor": "Tul Corporation / PowerColor",
      "vendor_id": "148C"
    }
  },
  "power_profile_modes": {
    "active": 5,
    "modes": {
      "0": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              1,
              0,
              4,
              800,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              1,
              0,
              3276800,
              -65536,
              -6553
            ]
          }
        ],
        "name": "BOOTUP_DEFAULT"
      },
      "1": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              0,
              1200,
              4,
              650,
              5242880,
              -3276,
              -65536
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              3,
              0,
              1310720,
              -6553,
              -6553
            ]
          }
        ],
        "name": "3D_FULL_SCREEN"
      },
      "2": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              1,
              0,
              3,
              0,
              5898240,
              -65536,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              1,
              0,
              1,
              0,
              3407872,
              -65536,
              -6553
            ]
          }
        ],
        "name": "POWER_SAVING"
      },
      "3": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              1,
              0,
              4,
              500,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              3,
              0,
              3473408,
              -65536,
              -6553
            ]
          }
        ],
        "name": "VIDEO"
      },
      "4": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              2,
              1000,
              1,
              0,
              3276800,
              0,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              3,
              0,
              1310720,
              -6553,
              -6553
            ]
          }
        ],
        "name": "VR"
      },
      "5": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              2,
              1000,
              1,
              0,
              3932160,
              0,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              3,
              0,
              1310720,
              -6553,
              -6553
            ]
          }
        ],
        "name": "COMPUTE"
      },
      "6": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              0,
              1200,
              4,
              0,
              655360,
              -3276,
              -65536
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              3,
              0,
              1310720,
              -6553,
              -6553
            ]
          }
        ],
        "name": "CUSTOM"
      },
      "7": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              0,
              1200,
              4,
              650,
              5242880,
              -3276,
              -65536
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              3,
              0,
              1310720,
              -6553,
              -6553
            ]
          }
        ],
        "name": "WINDOW_3D"
      }
    },
    "value_names": [
      "FPS",
      "MinActiveFreqType",
      "MinActiveFreq",
      "BoosterFreqType",
      "BoosterFreq",
      "PD_Data_limit_c",
      "PD_Data_error_coeff",
      "PD_Data_error_rate_coeff"
    ]
  },
  "power_states": {
    "core": [
      {
        "enabled": true,
        "index": 0,
        "value": 500
      },
      {
        "enabled": true,
        "index": 1,
        "value": 77
      },
      {
        "enabled": true,
        "index": 2,
        "value": 2254
      }
    ],
    "vram": [
      {
        "enabled": true,
        "index": 0,
        "value": 96
      },
      {
        "enabled": true,
        "index": 1,
        "value": 456
      },
      {
        "enabled": true,
        "index": 2,
        "value": 772
      },
      {
        "enabled": true,
        "index": 3,
        "value": 1218
      }
    ]
  },
  "stats": {
    "busy_percent": 2,
    "clockspeed": {
      "gpu_clockspeed": 2011,
      "vram_clockspeed": 456
    },
    "core_power_state": 1,
    "fan": {
      "control_enabled": false,
      "pmfw_info": {
        "acoustic_limit": {
          "allowed_range": [
            500,
            3300
          ],
          "current": 3000
        },
        "acoustic_target": {
          "allowed_range": [
            500,
            3300
          ],
          "current": 1200
        },
        "minimum_pwm": {
          "allowed_range": [
            35,
            100
          ],
          "current": 35
        },
        "target_temp": {
          "allowed_range": [
            25,
            110
          ],
          "current": 81
        },
        "zero_rpm_enable": true,
        "zero_rpm_temperature": {
          "allowed_range": [
            25,
            100
          ],
          "current": 50
        }
      },
      "pwm_current": 89,
      "pwm_max": 254,
      "pwm_min": 89,
      "speed_current": 868,
      "speed_max": 3300,
      "speed_min": 0,
      "temperature_range": [
        25,
        100
      ]
    },
    "memory_power_state": 1,
    "pcie_power_state": 2,
    "performance_level": "manual",
    "power": {
      "average": 32.0,
      "cap_current": 236.0,
      "cap_default": 236.0,
      "cap_max": 280.0,
      "cap_min": 212.0
    },
    "temps": {
      "edge": {
        "crit": 100.0,
        "crit_hyst": -273.1499938964844,
        "current": 24.0
      },
      "junction": {
        "crit": 110.0,
        "crit_hyst": -273.1499938964844,
        "current": 31.0
      },
      "mem": {
        "crit": 108.0,
        "crit_hyst": -273.1499938964844,
        "current": 48.0
      }
    },
    "voltage": {
      "gpu": 985
    },
    "vram": {}
  }
}
